### PR TITLE
[bitnami/ghost] Add Ghost vib tests

### PR DESCRIPTION
**Pull request by mdhont**
Signed-off-by: mdhont [michield@vmware.com](michield@vmware.com)

### Description of the change

The main objective of this PR is to publish our Bitnami Ghost container using VMware Image Builder. In order to do that, several changes are included:

- Increasing the existing test coverage of the asset by adding Goss tests.
- Update verify and publish VIB pipeline's definitions.

### Benefits

- Ensuring higher quality of the container catalog.
- Increased pool of assets completely handled by VMware Image Builder.

### Possible drawbacks

Automated tests could introduce additional flakiness to the CI/CD.

### Applicable issues

NA

### Additional information

Successful test run:
https://github.com/bitnami/containers/actions/runs/4798845018/jobs/8537675935

### CODE REVIEW

File: .vib/ghost/goss/ghost.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    exists: false
+  # Check if node executable directory is symlinked to default bin directory
+  /opt/bitnami/ghost/bin:
+    filetype: symlink
+    linked-to: /opt/bitnami/ghost/versions/{{ .Env.APP_VERSION }}/node_modules/.bin
+    exists: true
+  # Check permissions of libvips-cpp.so.42 have been set to 755 during compilation
+  /opt/bitnami/ghost/versions/{{ .Env.APP_VERSION }}/node_modules/sharp/vendor/8.12.2/linux-x64/lib/libvips-cpp.so.42:
+    exists: true
+    mode: "0755"
```
Comment (joancafom) [inline .vib/ghost/goss/ghost.yaml:17]: I'd avoid this check too. I don't think it adds much value and the path has many hardcoded versions which will be hard to maintain.

File: .vib/ghost/goss/vars.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+binaries:
+  - ghost
+  - mysql
+  - node
+  - python
+directories:
+  - paths:
+      - /home/ghost
+      - /opt/bitnami/ghost
```
[the remark this answers is not in the document]
Comment (mdhont) [reply .vib/ghost/goss/vars.yaml]: The permissions for "/opt/bitnami/ghost" do not seem to be explicitly set, (using the numerical notation I mean) so the test fails. (Permissions of that folder are 0777).

[the remark this answers is not in the document]
Comment (joancafom) [reply .vib/ghost/goss/vars.yaml]: The permissions of `/opt/bitnami/ghost` are explicitly set:

https://github.com/bitnami/containers/blob/6d597b976a0dbb9f10329820ef5f80cb75618d77/bitnami/ghost/5/debian-11/rootfs/opt/bitnami/scripts/ghost/postunpack.sh#L43

Moreover, the check should work:

```bash
$ stat $GHOST_BASE_DIR
  File: /opt/bitnami/ghost
  Size: 4096      	Blocks: 8          IO Block: 4096   directory
Device: 8dh/141d	Inode: 4073330     Links: 6
Access: (0777/drwxrwxrwx)  Uid: (  999/ UNKNOWN)   Gid: (    0/   ghost)
Access: 2023-05-17 14:04:41.896760757 +0000
Modify: 2023-05-17 13:07:34.000000000 +0000
Change: 2023-05-17 14:04:41.807761876 +0000
 Birth: 2023-05-17 14:04:25.346846632 +0000
```

File: .vib/ghost/goss/vars.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+binaries:
+  - mysql
+  - node
+  - python
+directories:
+  - paths:
+    - /home/ghost
+  - paths:
+    - /opt/bitnami/ghost
```
Comment (joancafom) [inline .vib/ghost/goss/vars.yaml:9]: The permissions of `/opt/bitnami/ghost` are explicitly set:

https://github.com/bitnami/containers/blob/6d597b976a0dbb9f10329820ef5f80cb75618d77/bitnami/ghost/5/debian-11/rootfs/opt/bitnami/scripts/ghost/postunpack.sh#L43

Moreover, the check should work:

```bash
$ stat $GHOST_BASE_DIR
  File: /opt/bitnami/ghost
  Size: 4096      	Blocks: 8          IO Block: 4096   directory
Device: 8dh/141d	Inode: 4073330     Links: 6
Access: (0777/drwxrwxrwx)  Uid: (  999/ UNKNOWN)   Gid: (    0/   ghost)
Access: 2023-05-17 14:04:41.896760757 +0000
Modify: 2023-05-17 13:07:34.000000000 +0000
Change: 2023-05-17 14:04:41.807761876 +0000
 Birth: 2023-05-17 14:04:25.346846632 +0000
```

**Review by joancafom — CHANGES_REQUESTED** [review]

**Review by joancafom — APPROVED** [review]

**Pull request merged**